src/osdc/Journaler.h

```diff
@@ -536,6 +538,7 @@ class Journaler {
   size_t get_journal_envelope_size() const { 
     return journal_stream.get_envelope_size(); 
   }
+  int check_isreadable();
```

Comment:
```suggestion
  int wait_for_readable();
```

Reply:
There has been a function named Journaler::wait_for_readable().
By the way,what new name is better, do you have any suggestions? :)

Reply:
Well that's interesting. I suggest fixing `Journaler::wait_for_readable` to account for these checks, instead of adding a new method.

Note that `PurgeQueue.cc` has the same code and needs fixed.

Reply:
Got it.
Do you think this method is OK? 
https://github.com/ceph/ceph/pull/48281#issuecomment-1297969787
(We keep wait_for_readable() uesd for PurgeQueue::_consume() )